lib/core/src/persist/receive.rs

```diff
@@ -204,17 +246,25 @@ impl Persister {
                         WHEN lockup_tx_id IS NULL THEN :lockup_tx_id
                         ELSE lockup_tx_id
                     END,
+                mrh_tx_id = 
+                    CASE
+                        WHEN mrh_tx_id IS NULL THEN :mrh_tx_id
+                        ELSE mrh_tx_id
+                    END,
+                payer_amount_sat = COALESCE(:mrh_amount_sat, payer_amount_sat),
+                receiver_amount_sat = COALESCE(:mrh_amount_sat, receiver_amount_sat),
```

Comment:
If a Receive Swap
- is paid via MRH, these two `COALESCE` lines will set the swap's `payer_amount_sat` and `receiver_amount_sat`
- if the swap is then also paid via Liquid swap, the swap's `payer` and `receiver` amounts remain unchanged, which means the Liquid swap's amounts will not be persisted

If it's paid twice but in the other order, the later MRH payment will overwrite the swap's initial payer/receiver amounts, which means they will be lost.

In both cases, the swap will be modified (because this is a `UPDATE receive_swaps`) in an inaccurate way.

Reply:
See https://github.com/breez/breez-sdk-liquid/pull/531#discussion_r1812628295